File: lib/ast/SourceFile.cpp
#include "hades/ast/SourceFile.h"

namespace hades {

SourceFile::SourceFile(SourceLocation location, Vec<const Declaration *> declarations)
    : m_location(location), m_declarations(std::move(declarations)) {}

auto SourceFile::declarations() const -> ArrayRef<const Declaration *> {
  return hades::ArrayRef<const Declaration *>(m_declarations);
}

auto SourceFile::location() const -> const SourceLocation & {
  return m_location;
}

} // namespace hades

File: hadesboot/src/main/kotlin/hadesc/ast/SourceFile.kt
package hadesc.ast

import hadesc.location.SourceLocation
import hadesc.qualifiedname.QualifiedName

data class SourceFile(
    override val location: SourceLocation,
    val id: Id,
    val moduleName: QualifiedName,
    val declarations: List<Declaration>,
) : ScopeTree {
    @JvmInline
    value class Id(val value: Int)
}
